examples/all-clusters-app/esp32/main/main.cpp

```diff
@@ -182,8 +186,19 @@ class EditAttributeListModel : public ListScreen::Model
         }
         else
         {
+            auto & name    = std::get<0>(attribute);
+            auto & cluster = std::get<0>(std::get<1>(std::get<1>(devices[deviceIndex])[endpointIndex])[i]);
+
             ESP_LOGI(TAG, "editing attribute as string: '%s' (%s)", value.c_str(), i == 0 ? "+" : "-");
             value = (value == "Closed") ? "Open" : "Closed";
+            ESP_LOGI(TAG, "name and cluster: '%s' (%s)", name.c_str(), cluster.c_str());
+            if (name == "State" && cluster == "Lock")
+            {
+                // update the doorlock attribute here
+                uint8_t attributeValue = value == "Closed" ? EMBER_ZCL_DOOR_LOCK_STATE_LOCKED : EMBER_ZCL_DOOR_LOCK_STATE_UNLOCKED;
+                emberAfWriteServerAttribute(DOOR_LOCK_SERVER_ENDPOINT, ZCL_DOOR_LOCK_CLUSTER_ID, ZCL_LOCK_STATE_ATTRIBUTE_ID,
+                                            (uint8_t *) &attributeValue, ZCL_INT8U_ATTRIBUTE_TYPE);
```

Comment:
nit: seems like this casting is unnecessary, the `attributeValue` is already of type `uint8_t`.